src/widgets.tsx

```diff
@@ -1,132 +1,244 @@
-            }\")`;
-            this.inject(cmd).then(() => {
-              if (resolve) {
-                resolve();
-              }
+
+  inject(code: string): any {
+    cell_utils
+      .insertAndRun(
+        this.commands,
+        this.notebook_panel,
+        this.notebook_panel.content.activeCellIndex,
+        code,
+        false
+      )
+      .then(result => {
+        return result;
+      })
+      .catch(error => {
+        console.log(error);
+      });
+  }
+
+  buildImportCommand(modules: string[], lazy: boolean): string {
+    let cmd: string = "";
+    //Check for lazy_imports modules first
+    let tmp_modules = modules;
+    let ind = modules.indexOf("lazy_import");
+
+    if (lazy) {
+      // Import lazy_imports if it's missing, before doing other imports
+      if (ind >= 0) {
+        tmp_modules.splice(ind, 1);
+        cmd = "import lazy_import";
+      }
+      // Import other modules using lazy import syntax
+      tmp_modules.forEach(module => {
+        cmd += `\n${module} = lazy_import.lazy_module("${module}")`;
+      });
+    } else {
+      // Remove lazy_imports from required modules if it is there
+      if (ind >= 0) {
+        tmp_modules.splice(ind, 1);
+      }
+      // Import modules
+      tmp_modules.forEach(module => {
+        cmd += `\nimport ${module}`;
+      });
+    }
+    return cmd;
+  }
+
+  // This will inject the required modules into the current notebook (if module not already imported)
+  injectRequiredModules(): Promise<number> {
+    console.log("Injecting required modules");
+    // Check if required modules are imported in notebook
+    var prom: Promise<number> = new Promise((resolve, reject) => {
+      cell_utils
+        .runAndDelete(this.commands, this.notebook_panel, CHECK_MODULES_CMD)
+        .then(output => {
+          let missing_modules: string[] = eval(output);
+          let cmd = this.buildImportCommand(missing_modules, true);
+          cell_utils
+            .insertAndRun(this.commands, this.notebook_panel, -1, cmd, false)
+            .then(result => {
+              resolve(result[0]);
+            })
+            .catch(error => {
+              reject(error);
             });
-          } else {
-            if (resolve) {
-              resolve();
-            }
-          }
+        })
+        .catch(error => {
+          console.log(error);
         });
-      });
     });
+
+    return prom;
   }
-  createNotebook(resolve: any, reject: any) {
-    this.commands
-      .execute("notebook:create-new", {
-        activate: true,
-        path: this.context.path,
-        preferredLanguage: this.context.model.defaultKernelLanguage
-      })
-      .then(notebook => {
-        this.prepNotebook(notebook, resolve);
-      })
-      .catch(error => {
-        if (reject) {
-          reject(error);
-        }
-      });
+
+  // Returns whether the current notebook is vcs ready (has required imports and vcs initialized)
+  isVCS_Ready(): boolean {
```

Comment:
this could be written as a getter as well.

Reply:
Thanks for the suggestion, done!